### ISSUE TITLE
[1.16.5 Mixin Crash] Unexpected null pointer exception trying to find ModList.isLoaded when using older generation hardware

### ISSUE THREAD

**Issue by NighthowlerStudios** [FalsehoodMC/Fabrication#584]
The only differences between all other players and this specific person's install is their CPU generation.  They are using a 4th gen Core i7-4770, as noted in the log.  However, all the other players I had have at least 9th gen or newer and no such crash has ever occurred for them.  This seems to be a thread race condition that causes forgery to fail first, then takes down Redstone Control with it in a domino style, something to do with the server config mixin.

tl;dr seems to be an incompatibility with older hardware and Redstone Control, but only if older hardware is present

is there a way I can disable whatever feature would cause your config mixins to fail?

is it a JVM specific bug, should I continue instructing the user to try updating java?

is it a forge bug?

https://pastebin.com/eDZ8L8t1

quote of importance:
```       
        Caused by: java.lang.NullPointerException
	at net.minecraftforge.fml.ModList.isLoaded(ModList.java:181) ~[?:?]
	at com.unascribed.fabrication.EarlyAgnos.isModLoaded(EarlyAgnos.java:27) ~[?:3.1.4+1.16]
```

**Comment by SFort** [comment]
this seames like a duplicate of #580 , which was fixed in 3.1.5
which atm is unreleased; if you're in a hurry you can compile the 3.0/1.16 branch (you'll need [forgeryTools](https://git.sleeping.town/unascribed/ForgeryTools), which uh you'd also need to compile)

**Comment by NighthowlerStudios** [comment]
understood.  sadly I have to wait for official curseforge release as it's only important for modpack use. Thanks!

as noted by #580 it must be because I have limited_runtime_configs enabled in my ini.  I will disable it now to avoid the crash

**Comment by NighthowlerStudios** [comment]
How soon are we expecting the patch?  Attempting to disable the feature supposedly at fault didn't fix the issue for my user.  I could try to compile but I'm busy today troubleshooting all the other projects I'm working on too.

is it possible that, unlike the mentioned issue, since I disabled the same config option and it didn't work, that it might be a separate issue?

**Comment by SFort** [comment]
it's not a conflicting feature.
it's just a part of forge that throws when called too early.
limited_runtime_configs is generally better for compat, since it reduces injections, but it can't do anything in this case.

the issue is the same and was fixed a while ago just never released.

**Comment by NighthowlerStudios** [comment]
"it's just a part of forge that throws when called too early." - yeah I figured as much I was just making sure.  it makes sense as well since forge's modloading order always does mixins before or alongside mod listing

what I mean to ask is there anything left on dev that needs to be done before the build can be uploaded to curse?  I won't be able to release the pack update I'm working on until forgery is at least able to load into a world.  As I mentioned before it works on my machine, but mine is miles faster than the problem machine

**Comment by SFort** [comment]
the patch couldof been released a month or so ago, and probably shouldof.
una does releases and i probably forgot to mention it at the time, and una operates on usa times so probably will be abit before a release.

**Comment by NighthowlerStudios** [comment]
alrighty, I'll keep tabs on it :) will close the issue once I have a build to test with

**Comment by NighthowlerStudios** [comment]
downloaded the update, still waiting on the tester to give it a try

**Comment by NighthowlerStudios** [comment]
late follow-up but the patch did fix my issue, thanks again.
